src/pdf_parser/pdf_utils/ocr.py

```diff
@@ -0,0 +1,487 @@
+    2. Layout parser detects a list of items as a single block but is unable to detect the individual items.
+    In this case, the best matches will be on what google classifies as blocks. So we want to merge the text
+    from the paragraphs into the blocks.
+
+    Regarding point 2, Google appears to be able to detect the individual items in a list more reliably than
+    layoutparser, a point we can explore in the future to get the best results by combining the two approaches.
+    For future usability, we also store block objects with paragraphs as sub-objects.
+
+    Args:
+        image: Image to extract document bounds from.
+
+    Returns:
+        List of GoogleBlocks, List of GoogleTextSegments, List of GoogleTextSegments, List of GoogleTextSegments
+    """
+
+    def _get_modal_string(string_list: List[str]) -> Optional[str]:
+        if len(string_list) > 0:
+            return max(set(block_languages), key=block_languages.count)
+        else:
+            return None
+
+    content = image_bytes(image)
+    client = vision.ImageAnnotatorClient()
+    image = types.Image(content=content)
+
+    # TODO: Handle errors. Hit a 503.
+    response = client.document_text_detection(image=image)
+    document = response.full_text_annotation
+
+    breaks = vision.enums.TextAnnotation.DetectedBreak.BreakType
+    paragraphs = []
+    lines = []
+    fully_structured_blocks = []
+    paragraph_text_segments = []
+    block_text_segments = []
+    for page in document.pages:
+        for block in page.blocks:
+            block_paras = []
+            block_para_coords = []
+            block_languages = []
+            para_confidences = []
+            for paragraph in block.paragraphs:
+                para_confidences.append(paragraph.confidence)
+                para = ""
+                line = ""
+                para_languages = (
+                    []
+                )  # languages stored at word level. Detect then take mode.
+                for word in paragraph.words:
+                    try:
+                        lang = word.property.detected_languages[0].language_code
+                    except IndexError:
+                        lang = None
+                    para_languages.append(lang)
+                    block_languages.append(lang)
+                    for symbol in word.symbols:
+                        line += symbol.text
+                        if symbol.property.detected_break.type == breaks.SPACE:
+                            line += " "
+                        if symbol.property.detected_break.type == breaks.EOL_SURE_SPACE:
+                            line += " "
+                            lines.append(line)
+                            para += line
+                            line = ""
+                        if symbol.property.detected_break.type == breaks.LINE_BREAK:
+                            lines.append(line)
+                            para += line
+                            line = ""
```

Comment:
This logic makes sense with a close read but would be good to add comments to these if statements about what they're doing. E.g. `# Start new line in same paragraph`.
Potentially worth merging the logic for `breaks.LINE_BREAK` and `breaks.EOL_SURE_SPACE` as the first seems to do what the second does, plus adding a space to the end of the line?